Integrations/src/test/java/io/deephaven/integrations/learn/GathererTest.java

```diff
@@ -0,0 +1,113 @@
+package io.deephaven.integrations.learn;
+
+import io.deephaven.db.v2.InMemoryTable;
+import io.deephaven.db.v2.sources.ColumnSource;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import java.util.Arrays;
+
+public class GathererTest {
+
+    private static InMemoryTable table;
+
+    @BeforeClass
+    public static void setup() {
+        table = new InMemoryTable(
+                new String[] {"bool1", "bool2",
+                        "byte1", "byte2",
+                        "short1", "short2",
+                        "int1", "int2",
+                        "long1", "long2",
+                        "float1", "float2",
+                        "double1", "double2"},
+                new Object[] {
+                        new boolean[] {true, true, false, false},
+                        new boolean[] {true, false, true, false},
+                        new byte[] {(byte)1, (byte)2, (byte)3, (byte)4},
+                        new byte[] {(byte)5, (byte)6, (byte)7, (byte)8},
+                        new short[] {(short)-1, (short)-2, (short)-3, (short)-4},
+                        new short[] {(short)-5, (short)-6, (short)-7, (short)-8},
+                        new int[] {100, 200, -100, -200},
+                        new int[] {-300, -400, 300, 400},
+                        new long[] {1L, 100L, 10000L, 1000000L},
+                        new long[] {9L, 999L, 99999L, 9999999L},
+                        new float[] {3.14F, 2.73F, 1.5F, 0.63F},
+                        new float[] {0.1F, 0.2F, 0.3F, 0.4F},
+                        new double[] {3.14, 2.73, 1.5, 0.63},
+                        new double[] {0.1, 0.2, 0.3, 0.4}
+                }
+        );
+    }
+
+    public static IndexSet makeIndexSet(final InMemoryTable t) {
+        final IndexSet is = new IndexSet(t.intSize());
+
+        for (long idx : t.getIndex()) {
+            is.add(idx);
+        }
+
+        return is;
+    }
+
+    @Test
+    public void gatherTestMethod() {
```

Comment:
There should be a separate test method for each method.  Right now, a failure in one type will mask failures in other types, which increases the time to create a fix.